test/diagnostics/Localization/en_localization_parser_lookup.swift

```diff
@@ -1,9 +1,18 @@
 // RUN: %target-typecheck-verify-swift -localization-path %S/Inputs -locale en -enable-parser-lookup
+// RUN: not %target-swift-frontend -debug-diagnostic-names -localization-path %S/Inputs -locale en -enable-parser-lookup -typecheck %s 2>&1 | %FileCheck %s --check-prefix=CHECK_NAMES
```

Comment:
I added tests here as well to make sure that we can print diagnostic names from the `YAMLLocalizationProducer` (not the serialized format)